README.md
nette2
======

tutorials


nette/history.txt
2013-07-11 (dc83a21): Released version 2.0.11
2013-07-11 (2ed3fd4): whitespace, used common (and verbose) indentation in switch statements
2013-07-11 (a9152cd): Route: apply filter out even on non-scalar parameters.
2013-07-11 (993bed2): SimpleRouter: fixed array to string conversion if presenter key contains array
2013-07-11 (7e4d1fc): Using only array_values from $args array, when we use IN in WHERE statment.
2013-07-09 (fa5424c): Session: Removed dead code
2013-07-09 (de32bbf): Mail: workaround for ignored Return-Path header (thx frosty22)
2013-07-09 (52a9db1): Mail\Message: fixed never working getReturnPath()
2013-07-09 (7264e0f): Route: fixed returning invalid application request with presenter as array
2013-07-09 (b07896e): requires Nette\Tester 0.9.*
2013-07-09 (7ddba10): Templating: helper nl2br recognises xhtml settings on class Nette\Utils\Html
2013-07-08 (9c30bd4): Debugger: document.body is accessed in <body>
2013-07-08 (e31f864): Session: restore error handler only once when starting fails
2013-07-06 (11630e1): Debugger Bar: added 50ms timeout before panel is displayed (fixed)
2013-07-06 (6a14bcb): Latte: fixed HTML comments escaping. All dashes are removed, for sure.
2013-07-06 (6f03685): typos
2013-07-06 (a79686b): Html: added empty (or void) element <track>
2013-07-05 (8288142): used @return self for fluent interfaces
2013-07-05 (cf2e2a2): tests: fixed bug #48724 in PHP 5.4.17
2013-07-02 (7a14c2d): whitespace (3 empty lines changed to 2)
2013-07-02 (10f72ad): Forms: SelectBox: fixed missing attribute value when option is Html object
2013-06-28 (ffa3770): Database: fixed support for tables without primary-key
2013-06-23 (556b9af): tests: fixed bug #48724 in PHP 5.5
2013-06-23 (25cb01a): Json::encode() fixed PHP 5.5 compatibility
2013-06-20 (8472fc7): tests: travis: removed allow_failures for PHP 5.5
2013-06-20 (e2e94c1): tests: image rotate test checks only images dimension, because in PHP 5.5 rotation algorithm has been changed
2013-06-20 (d766806): tests: removed application/octet-stream MIME detection because can vary on different platforms (fixes application/CDFV2-corrupt detection)
2013-06-20 (d218492): Latte: added n:name, forward compatible alias for n:input
2013-06-20 (716d7ae): Forms: fixed cutting strings according to Form::LENGTH
2013-06-20 (15c86be): Debugger: in MSIE sends error 503 instead of 500 to ensure error page is displayed
2013-06-17 (6a1daf7): Debugger bluescreen: removed z-index
2013-06-17 (739f9e4): added alias Nette\Config\Configurator -> Nette\Configurator
2013-06-17 (fcedc22): whitespace
2013-06-17 (9da9e22): test refactoring, used new assertions
2013-06-17 (708fcbb): magic service accessors replaced with getService()
2013-06-17 (331b6bc): used func_get_args() as argument
2013-06-17 (45996dc): tests: SqlBuilder.addWhere() correction
2013-06-17 (cbe5a9c): tests: typo
2013-06-17 (114e2a8): tests: some test renamed
2013-06-17 (534db8b): tests: simplified database tests, added requote()
2013-06-17 (b026259): Database: DiscoveredReflection::reloadAllForeignKeys() resets old structure
2013-06-17 (69c986d): Database: replaced Connection::exec() with query()
2013-06-17 (8309a49): Database: SqliteDriver::getTables() skips internal tables
2013-06-17 (74dc17a): Database: added name and DSN to ConnectionPanel
2013-06-17 (abc42f8): Database: row normalization is required for SQLite3 too (it's not needed for table selects, but is needed for view selects) [Closes #1054]
2013-06-17 (44ee595): Database: fixed cathing word endings in SqlBuilder::addWhere()
2013-06-17 (59a9c0f): Database: SqlPreprocessor throws exception when parameter is missing.
2013-06-17 (1779b6f): Database: fixed support for Traversable in SqlPreprocessor
2013-06-17 (ccbcb7e): test: added Database tests
2013-06-17 (c5f7b07): Database: fixed adding multiline condition
2013-06-17 (2276498): Database: typo
2013-06-17 (13c4dc4): Database: iterate over exact count of columns in Statement::detectColumnTypes()
2013-06-17 (75c988b): Database: renamed test to SqlBuilder.tryDelimite()
2013-06-17 (ce2d2fb): Fixed test
2013-06-17 (dd3cd21): Database: fixed auto adding operator
2013-06-17 (5b0eecc): Database: fixed operator control in SqlBuilder [Closes #996]
2013-06-17 (cf685f6): NetteExtension: only PDO:: constants in database options are converted to real value [Closes #700]
2013-06-17 (4c8f6d8): Database: removed useless condition
2013-06-17 (d641ac7): Database: sort is quicker than array_diff
2013-06-17 (443ea6a): Database: fixed work with undefined primary key [Refs #288]

nette/sandbox/temp/cache/_Nette.Configurator/_-492803ce0021ff2383f33187831644ca.php
<?php //netteCache[01]008565a:2:{s:4:"time";s:21:"0.81922500 1374191062";s:9:"callbacks";a:49:{i:0;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:61:"D:\_projects\Tran\nette2\nette\sandbox\app\config\config.neon";i:2;i:1374191059;}i:1;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\app\config\config.local.neon";i:2;i:1374190490;}i:2;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:84:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Config\Extensions\PhpExtension.php";i:2;i:1374190491;}i:3;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:90:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Config\Extensions\ConstantsExtension.php";i:2;i:1374190491;}i:4;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:86:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Config\Extensions\NetteExtension.php";i:2;i:1374190491;}i:5;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\common\Object.php";i:2;i:1374190491;}i:6;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:79:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Caching\Storages\IJournal.php";i:2;i:1374190491;}i:7;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:82:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Caching\Storages\FileJournal.php";i:2;i:1374190491;}i:8;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:70:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Caching\IStorage.php";i:2;i:1374190491;}i:9;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:82:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Caching\Storages\FileStorage.php";i:2;i:1374190491;}i:10;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:73:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\RequestFactory.php";i:2;i:1374190491;}i:11;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\IRequest.php";i:2;i:1374190491;}i:12;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\Request.php";i:2;i:1374190491;}i:13;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:68:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\IResponse.php";i:2;i:1374190491;}i:14;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\Response.php";i:2;i:1374190491;}i:15;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\Context.php";i:2;i:1374190491;}i:16;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\Session.php";i:2;i:1374190491;}i:17;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:75:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Security\IUserStorage.php";i:2;i:1374190491;}i:18;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:70:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Http\UserStorage.php";i:2;i:1374190491;}i:19;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Security\User.php";i:2;i:1374190491;}i:20;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:77:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Application\Application.php";i:2;i:1374190490;}i:21;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:83:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Application\IPresenterFactory.php";i:2;i:1374190490;}i:22;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:82:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Application\PresenterFactory.php";i:2;i:1374190490;}i:23;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:73:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Application\IRouter.php";i:2;i:1374190490;}i:24;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Mail\IMailer.php";i:2;i:1374190491;}i:25;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:73:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Mail\SendmailMailer.php";i:2;i:1374190491;}i:26;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:71:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\DI\NestedAccessor.php";i:2;i:1374190491;}i:27;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:73:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Database\Connection.php";i:2;i:1374190491;}i:28;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:63:"D:\_projects\Tran\nette2\nette\sandbox\app\model\Repository.php";i:2;i:1374190962;}i:29;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\app\model\UserRepository.php";i:2;i:1374190490;}i:30;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\app\model\ListRepository.php";i:2;i:1374190490;}i:31;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\app\model\TaskRepository.php";i:2;i:1374190490;}i:32;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\app\router\RouterFactory.php";i:2;i:1374190490;}i:33;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:77:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Security\IAuthenticator.php";i:2;i:1374190491;}i:34;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\app\model\Authenticator.php";i:2;i:1374190490;}i:35;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:76:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\common\FreezableObject.php";i:2;i:1374190491;}i:36;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:71:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\common\IFreezable.php";i:2;i:1374190491;}i:37;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\DI\IContainer.php";i:2;i:1374190491;}i:38;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\DI\Container.php";i:2;i:1374190491;}i:39;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:90:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Application\Diagnostics\RoutingPanel.php";i:2;i:1374190490;}i:40;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:64:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Forms\Form.php";i:2;i:1374190491;}i:41;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:69:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\common\Callback.php";i:2;i:1374190491;}i:42;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:67:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Caching\Cache.php";i:2;i:1374190491;}i:43;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:94:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Database\Reflection\DiscoveredReflection.php";i:2;i:1374190491;}i:44;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Latte\Engine.php";i:2;i:1374190491;}i:45;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:66:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Mail\Message.php";i:2;i:1374190491;}i:46;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:77:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Templating\FileTemplate.php";i:2;i:1374190491;}i:47;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:73:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Templating\Template.php";i:2;i:1374190491;}i:48;a:3:{i:0;a:2:{i:0;s:19:"Nette\Caching\Cache";i:1;s:9:"checkFile";}i:1;s:84:"D:\_projects\Tran\nette2\nette\sandbox\libs\Nette\Security\Diagnostics\UserPanel.php";i:2;i:1374190491;}}}?><?php
// source: D:\_projects\Tran\nette2\nette\sandbox\app/config/config.neon development
// source: D:\_projects\Tran\nette2\nette\sandbox\app/config/config.local.neon 

/**
 * @property Nette\Application\Application $application
 * @property Authenticator $authenticator
 * @property Nette\Caching\Storages\FileStorage $cacheStorage
 * @property Nette\DI\NestedAccessor $constants
 * @property Nette\DI\Container $container
 * @property Nette\Http\Request $httpRequest
 * @property Nette\Http\Response $httpResponse
 * @property Todo\ListRepository $listRepository
 * @property SystemContainer_nette $nette
 * @property Nette\DI\NestedAccessor $php
 * @property Nette\Application\IRouter $router
 * @property RouterFactory $routerFactory
 * @property Nette\Http\Session $session
 * @property Todo\TaskRepository $taskRepository
 * @property Nette\Security\User $user
 * @property Todo\UserRepository $userRepository
 */
class SystemContainer extends Nette\DI\Container
{

	public $classes = array(
		'nette\\object' => FALSE, //: nette.cacheJournal, cacheStorage, nette.httpRequestFactory, httpRequest, httpResponse, nette.httpContext, session, nette.userStorage, user, application, nette.presenterFactory, nette.mailer, nette.database, userRepository, listRepository, taskRepository, authenticator, container,
		'nette\\caching\\storages\\ijournal' => 'nette.cacheJournal',
		'nette\\caching\\storages\\filejournal' => 'nette.cacheJournal',
		'nette\\caching\\istorage' => 'cacheStorage',
		'nette\\caching\\storages\\filestorage' => 'cacheStorage',
		'nette\\http\\requestfactory' => 'nette.httpRequestFactory',
		'nette\\http\\irequest' => 'httpRequest',
		'nette\\http\\request' => 'httpRequest',
		'nette\\http\\iresponse' => 'httpResponse',
		'nette\\http\\response' => 'httpResponse',
		'nette\\http\\context' => 'nette.httpContext',
		'nette\\http\\session' => 'session',
		'nette\\security\\iuserstorage' => 'nette.userStorage',
		'nette\\http\\userstorage' => 'nette.userStorage',
		'nette\\security\\user' => 'user',
		'nette\\application\\application' => 'application',
		'nette\\application\\ipresenterfactory' => 'nette.presenterFactory',
		'nette\\application\\presenterfactory' => 'nette.presenterFactory',
		'nette\\application\\irouter' => 'router',
		'nette\\mail\\imailer' => 'nette.mailer',
		'nette\\mail\\sendmailmailer' => 'nette.mailer',
		'nette\\di\\nestedaccessor' => 'nette.database',
		'pdo' => 'nette.database.default',
		'nette\\database\\connection' => 'nette.database.default',
		'todo\\repository' => FALSE, //: userRepository, listRepository, taskRepository,
		'todo\\userrepository' => 'userRepository',
		'todo\\listrepository' => 'listRepository',
		'todo\\taskrepository' => 'taskRepository',
		'routerfactory' => 'routerFactory',
		'nette\\security\\iauthenticator' => 'authenticator',
		'authenticator' => 'authenticator',
		'nette\\freezableobject' => 'container',
		'nette\\ifreezable' => 'container',
		'nette\\di\\icontainer' => 'container',
		'nette\\di\\container' => 'container',
	);

	public $meta = array();


	public function __construct()
	{
		parent::__construct(array(
			'appDir' => 'D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app',
			'wwwDir' => 'D:/_projects/Tran/nette2/nette/sandbox/www',
			'debugMode' => TRUE,
			'productionMode' => FALSE,
			'environment' => 'development',
			'consoleMode' => FALSE,
			'container' => array(
				'class' => 'SystemContainer',
				'parent' => 'Nette\\DI\\Container',
			),
			'tempDir' => 'D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app/../temp',
		));
	}


	/**
	 * @return Nette\Application\Application
	 */
	protected function createServiceApplication()
	{
		$service = new Nette\Application\Application($this->getService('nette.presenterFactory'), $this->getService('router'), $this->getService('httpRequest'), $this->getService('httpResponse'));
		$service->catchExceptions = FALSE;
		$service->errorPresenter = 'Error';
		Nette\Application\Diagnostics\RoutingPanel::initializePanel($service);
		Nette\Diagnostics\Debugger::$bar->addPanel(new Nette\Application\Diagnostics\RoutingPanel($this->getService('router'), $this->getService('httpRequest')));
		return $service;
	}


	/**
	 * @return Authenticator
	 */
	protected function createServiceAuthenticator()
	{
		$service = new Authenticator($this->getService('userRepository'));
		return $service;
	}


	/**
	 * @return Nette\Caching\Storages\FileStorage
	 */
	protected function createServiceCacheStorage()
	{
		$service = new Nette\Caching\Storages\FileStorage('D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app/../temp/cache', $this->getService('nette.cacheJournal'));
		return $service;
	}


	/**
	 * @return Nette\DI\NestedAccessor
	 */
	protected function createServiceConstants()
	{
		$service = new Nette\DI\NestedAccessor($this, 'constants');
		return $service;
	}


	/**
	 * @return Nette\DI\Container
	 */
	protected function createServiceContainer()
	{
		return $this;
	}


	/**
	 * @return Nette\Http\Request
	 */
	protected function createServiceHttpRequest()
	{
		$service = $this->getService('nette.httpRequestFactory')->createHttpRequest();
		if (!$service instanceof Nette\Http\Request) {
			throw new Nette\UnexpectedValueException('Unable to create service \'httpRequest\', value returned by factory is not Nette\\Http\\Request type.');
		}
		return $service;
	}


	/**
	 * @return Nette\Http\Response
	 */
	protected function createServiceHttpResponse()
	{
		$service = new Nette\Http\Response;
		return $service;
	}


	/**
	 * @return Todo\ListRepository
	 */
	protected function createServiceListRepository()
	{
		$service = new Todo\ListRepository($this->getService('nette.database.default'));
		return $service;
	}


	/**
	 * @return Nette\DI\NestedAccessor
	 */
	protected function createServiceNette()
	{
		$service = new Nette\DI\NestedAccessor($this, 'nette');
		return $service;
	}


	/**
	 * @return Nette\Forms\Form
	 */
	public function createNette__basicForm()
	{
		$service = new Nette\Forms\Form;
		return $service;
	}


	/**
	 * @return Nette\Callback
	 */
	protected function createServiceNette__basicFormFactory()
	{
		$service = new Nette\Callback($this, 'createNette__basicForm');
		return $service;
	}


	/**
	 * @return Nette\Caching\Cache
	 */
	public function createNette__cache($namespace = NULL)
	{
		$service = new Nette\Caching\Cache($this->getService('cacheStorage'), $namespace);
		return $service;
	}


	/**
	 * @return Nette\Callback
	 */
	protected function createServiceNette__cacheFactory()
	{
		$service = new Nette\Callback($this, 'createNette__cache');
		return $service;
	}


	/**
	 * @return Nette\Caching\Storages\FileJournal
	 */
	protected function createServiceNette__cacheJournal()
	{
		$service = new Nette\Caching\Storages\FileJournal('D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app/../temp');
		return $service;
	}


	/**
	 * @return Nette\DI\NestedAccessor
	 */
	protected function createServiceNette__database()
	{
		$service = new Nette\DI\NestedAccessor($this, 'nette.database');
		return $service;
	}


	/**
	 * @return Nette\Database\Connection
	 */
	protected function createServiceNette__database__default()
	{
		$service = new Nette\Database\Connection('mysql:host=localhost;dbname=quickstart', 'root', NULL, NULL);
		$service->setCacheStorage($this->getService('cacheStorage'));
		Nette\Diagnostics\Debugger::$blueScreen->addPanel('Nette\\Database\\Diagnostics\\ConnectionPanel::renderException');
		$service->setDatabaseReflection(new Nette\Database\Reflection\DiscoveredReflection($this->getService('cacheStorage')));
		$service->onQuery[] = array(
			$this->getService('nette.database.defaultConnectionPanel'),
			'logQuery',
		);
		return $service;
	}


	/**
	 * @return Nette\Database\Diagnostics\ConnectionPanel
	 */
	protected function createServiceNette__database__defaultConnectionPanel()
	{
		$service = new Nette\Database\Diagnostics\ConnectionPanel;
		$service->explain = TRUE;
		$service->name = 'default';
		Nette\Diagnostics\Debugger::$bar->addPanel($service);
		return $service;
	}


	/**
	 * @return Nette\Http\Context
	 */
	protected function createServiceNette__httpContext()
	{
		$service = new Nette\Http\Context($this->getService('httpRequest'), $this->getService('httpResponse'));
		return $service;
	}


	/**
	 * @return Nette\Http\RequestFactory
	 */
	protected function createServiceNette__httpRequestFactory()
	{
		$service = new Nette\Http\RequestFactory;
		$service->setEncoding('UTF-8');
		return $service;
	}


	/**
	 * @return Nette\Latte\Engine
	 */
	public function createNette__latte()
	{
		$service = new Nette\Latte\Engine;
		return $service;
	}


	/**
	 * @return Nette\Callback
	 */
	protected function createServiceNette__latteFactory()
	{
		$service = new Nette\Callback($this, 'createNette__latte');
		return $service;
	}


	/**
	 * @return Nette\Mail\Message
	 */
	public function createNette__mail()
	{
		$service = new Nette\Mail\Message;
		$service->setMailer($this->getService('nette.mailer'));
		return $service;
	}


	/**
	 * @return Nette\Callback
	 */
	protected function createServiceNette__mailFactory()
	{
		$service = new Nette\Callback($this, 'createNette__mail');
		return $service;
	}


	/**
	 * @return Nette\Mail\SendmailMailer
	 */
	protected function createServiceNette__mailer()
	{
		$service = new Nette\Mail\SendmailMailer;
		return $service;
	}


	/**
	 * @return Nette\Application\PresenterFactory
	 */
	protected function createServiceNette__presenterFactory()
	{
		$service = new Nette\Application\PresenterFactory('D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app', $this);
		return $service;
	}


	/**
	 * @return Nette\Templating\FileTemplate
	 */
	public function createNette__template()
	{
		$service = new Nette\Templating\FileTemplate;
		$service->registerFilter($this->createNette__latte());
		$service->registerHelperLoader('Nette\\Templating\\Helpers::loader');
		return $service;
	}


	/**
	 * @return Nette\Caching\Storages\PhpFileStorage
	 */
	protected function createServiceNette__templateCacheStorage()
	{
		$service = new Nette\Caching\Storages\PhpFileStorage('D:\\_projects\\Tran\\nette2\\nette\\sandbox\\app/../temp/cache', $this->getService('nette.cacheJournal'));
		return $service;
	}


	/**
	 * @return Nette\Callback
	 */
	protected function createServiceNette__templateFactory()
	{
		$service = new Nette\Callback($this, 'createNette__template');
		return $service;
	}


	/**
	 * @return Nette\Http\UserStorage
	 */
	protected function createServiceNette__userStorage()
	{
		$service = new Nette\Http\UserStorage($this->getService('session'));
		return $service;
	}


	/**
	 * @return Nette\DI\NestedAccessor
	 */
	protected function createServicePhp()
	{
		$service = new Nette\DI\NestedAccessor($this, 'php');
		return $service;
	}


	/**
	 * @return Nette\Application\IRouter
	 */
	protected function createServiceRouter()
	{
		$service = $this->getService('routerFactory')->createRouter();
		if (!$service instanceof Nette\Application\IRouter) {
			throw new Nette\UnexpectedValueException('Unable to create service \'router\', value returned by factory is not Nette\\Application\\IRouter type.');
		}
		return $service;
	}


	/**
	 * @return RouterFactory
	 */
	protected function createServiceRouterFactory()
	{
		$service = new RouterFactory;
		return $service;
	}


	/**
	 * @return Nette\Http\Session
	 */
	protected function createServiceSession()
	{
		$service = new Nette\Http\Session($this->getService('httpRequest'), $this->getService('httpResponse'));
		$service->setExpiration('14 days');
		return $service;
	}


	/**
	 * @return Todo\TaskRepository
	 */
	protected function createServiceTaskRepository()
	{
		$service = new Todo\TaskRepository($this->getService('nette.database.default'));
		return $service;
	}


	/**
	 * @return Nette\Security\User
	 */
	protected function createServiceUser()
	{
		$service = new Nette\Security\User($this->getService('nette.userStorage'), $this);
		Nette\Diagnostics\Debugger::$bar->addPanel(new Nette\Security\Diagnostics\UserPanel($service));
		return $service;
	}


	/**
	 * @return Todo\UserRepository
	 */
	protected function createServiceUserRepository()
	{
		$service = new Todo\UserRepository($this->getService('nette.database.default'));
		return $service;
	}


	public function initialize()
	{
		date_default_timezone_set('Europe/Prague');
		Nette\Caching\Storages\FileStorage::$useDirectories = TRUE;

		$this->getService("session")->exists() && $this->getService("session")->start();
		header('X-Frame-Options: SAMEORIGIN');
	}

}



/**
 * @property Nette\Database\Connection $default
 * @property Nette\Database\Diagnostics\ConnectionPanel $defaultConnectionPanel
 */
class SystemContainer_nette_database
{



}



/**
 * @method Nette\Forms\Form createBasicForm()
 * @property Nette\Callback $basicFormFactory
 * @method Nette\Caching\Cache createCache()
 * @property Nette\Callback $cacheFactory
 * @property Nette\Caching\Storages\FileJournal $cacheJournal
 * @property SystemContainer_nette_database $database
 * @property Nette\Http\Context $httpContext
 * @method Nette\Latte\Engine createLatte()
 * @property Nette\Callback $latteFactory
 * @method Nette\Mail\Message createMail()
 * @property Nette\Callback $mailFactory
 * @property Nette\Mail\SendmailMailer $mailer
 * @property Nette\Application\PresenterFactory $presenterFactory
 * @method Nette\Templating\FileTemplate createTemplate()
 * @property Nette\Caching\Storages\PhpFileStorage $templateCacheStorage
 * @property Nette\Callback $templateFactory
 * @property Nette\Http\UserStorage $userStorage
 */
class SystemContainer_nette
{



}
